contracts/SpokePool.sol

```diff
@@ -43,6 +45,15 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     // Origin token to destination token routings can be turned on or off.
     mapping(address => mapping(uint256 => bool)) public enabledDepositRoutes;
 
+    struct RelayerRefund {
+        // Merkle root of relayer refunds.
+        bytes32 distributionRoot;
```

Comment:
Hm so you think this could work by the `HubPool` relaying all leafs to the `SpokePool` here and storing them on-chain, as opposed to just relaying the root. @chrismaree wdyt?

Reply:
I dont think @mrice32  is proposing the HubPool passing the leafs; it cant do that in a efficient way. I _think_ @mrice32 is proposing a technique where you can execute multiple leafs at once in a compressed structure and prove they are within the provided root. @mrice32 what you describe would fit into the execution stage of the process, right?

Reply:
How do you imagine we improve on just making the contract multicall and assuming `distributeRelayerRefund` is called multiple times in one batched txn?

Reply:
I wanna be super clear that I don't think we should do this now, and I think before we add this complexity, we should try to better understand the amount of calldata/gas savings we'd see from something like this with different tree sizes.

Agree @chrismaree. I'm not suggesting the HubPool do anything different since any added data anywhere on mainnet is going to be expensive.

If you did multicall, you would effectively be passing:

[{distributionStruct1, proofArray1}, {distributionStruct2, proofArray2}...]

What I'm suggesting is an additional method that allows you to pass _all_ leaves this way:

[{distriubtionStruct1}, {distributionStruct2}, ...]

If we get the logic right in the contract, the leaf data _could be used_ to generate the proofs so you wouldn't need to pass them in, thereby reducing calldata.